scripts/use-react-version.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable no-console */
+/**
+ * Given the dist tag fetch the corresponding
+ * version and make sure this version is used throughout the repository.
+ *
+ * If you work on this file:
+ * WARNING: This script can only use built-in modules since it has to run before
+ * `yarn install`
+ */
+const childProcess = require('child_process');
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const { promisify } = require('util');
+
+const exec = promisify(childProcess.exec);
+
+// packages published from the react monorepo using the same version
+const reactPackageNames = ['react', 'react-dom', 'react-is', 'react-test-renderer', 'scheduler'];
+const devDependenciesPackageNames = ['@mnajdova/enzyme-adapter-react-18', '@testing-library/react'];
+
+// if we need to support more versions we will need to add new mapping here
+const additionalVersionsMappings = {
+  17: {
+    scheduler: '^0.20.0',
+    '@mnajdova/enzyme-adapter-react-18': 'npm:@eps1lon/enzyme-adapter-react-17',
+    '@testing-library/react': '^12.1.0',
+  },
+};
+
+async function main(version) {
+  if (typeof version !== 'string') {
+    throw new TypeError(`expected version: string but got '${version}'`);
+  }
+
+  if (version === 'stable') {
+    console.log('Nothing to do with stable');
+    return;
+  }
+
+  const packageJsonPath = path.resolve(__dirname, '../package.json');
+  const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, { encoding: 'utf8' }));
+
+  let majorVersion = null;
+
+  if (version.startsWith('^') || version.startsWith('~') || !Number.isNaN(version.charAt(0))) {
+    majorVersion = version.replace('^', '').replace('~', '').split('.')[0];
```

Comment:
If this ever becomes too tedious to maintain, `npm view --json react@^17.0.0 version` might be better to resolve to an exact (SemVer compatible) version. Extracting the MAJOR from that is probably simpler.

Reply:
Yep, this helps a lot, it's much cleaner now. Also, I've used this command (with the `dependencies` option, to get the `scheduler` version so that I don't have to hard code it for each version).

Reply:
The output on CI and locally is different, I'll revert the logic for extracting the major.